sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1571,6 +1602,89 @@ a normal SYCL command-group submission.
 associated with the graph node resulting from this command-group submission is
 different from the one with which the dynamic_parameter was created.
 
+|===
+
+=== Dynamic Events [[dynamic-events]]
+
+[source,c++]
+----
+namespace ext::oneapi::experimental{
+  class dynamic_event{
+    dynamic_event();
+    dynamic_event(const event& ev);
+
+    void update(const event& ev);
+  };
+}
+----
+
+Dynamic events represent SYCL events from outside of a given `command_graph`
+(either obtained from normal SYCL operations or from another `command_graph`)
+that nodes in that graph may depend on.  Dynamic events also allow for these
+dependent events to be updated between graph executions.
+
+Dynamic events can be used to add dependencies to a graph node in the same way
+that regular SYCL events can, by passing them as parameters to
+`handler::depends_on()` inside the CGF which represents the node.
+
+[source,c++]
+----
+// Obtain an event from a normal queue submission
+event OutsideEvent = queue.submit(...);
+
+// Create a dynamic event to wrap that event
+ext::oneapi::experimental::dynamic_event DynEvent {OutsideEvent};
+
+// Add a graph node which depends on that dynamic event
+Graph.add([&](handler& CGH){
+  CGH.depends_on(DynEvent);
+  CGH.parallel_for(...);
+});
+----
+Dynamic events created with an event from a `command_graph` cannot then be
+associated with other nodes in that same graph as this would change the shape of
+the graph. Attempting to call `handler::depends_on()` with such a
+`dynamic_event` in that situation will result in an error.
+
+Dynamic events can be created with no event but must be updated with a valid
+event before any executable graph which depends on that event is executed.
+Failing to do so will result in an error.
+
+Table {counter: tableNumber}. Member functions of the `dynamic_event` class.
+[cols="2a,a"]
+|===
+|Member function|Description
+
+|
+[source,c++]
+----
+dynamic_event();
+----
+
+| Constructs a default `dynamic_event` which is not associated with any SYCL event.
+
+|
+[source,c++]
+----
+dynamic_event(const event& ev);
+----
+
+| Constructs a `dynamic_event` which is associated with the SYCL event `ev`.
+
+
+|
+[source,c++]
+----
+void update(const event& ev);
+----
+
+| Updates the SYCL event associated with `dynamic_event`. This update will be
+reflected immediately in the associated modifiable graph nodes. An executable
+graph can then be updated to reflect these new event dependencies using
+<<executable-graph-update, Executable Graph Update>>.
```

Comment:
Not sure if that Executable Graph Update intro text needs updated - it's says "The only type of nodes that are currently supported for updating in a graph are kernel execution nodes." Which I don't think is a restriction for this feature.

Reply:
I've updated the wording there to specify that all node types can have dynamic events updated.